tests/linalg_test.py

```diff
@@ -1131,5 +1131,31 @@ def testIssue2131(self, n, dtype):
                             check_dtypes=True)
     self._CompileAndCheck(jsp_fun, args_maker_zeros, check_dtypes=True)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name":
+       "_lhs={}_rhs={}".format(
+           jtu.format_shape_dtype_string(lhs_shape, dtype),
+           jtu.format_shape_dtype_string(rhs_shape, dtype)),
+       "lhs_shape": lhs_shape,
+       "rhs_shape": rhs_shape, "dtype": dtype, "rng_factory": rng_factory}
+      for lhs_shape, rhs_shape in [
+          ((2, 4), (2, 2)),
+          ((5, 4), (5, 2)),
+          ((4, 4), (4, 1))
+      ]
+      for dtype in float_types
+      for rng_factory in [jtu.rand_default]))
+  def testLstsq(self, lhs_shape, rhs_shape, dtype, rng_factory):
+    _skip_if_unsupported_type(dtype)
+    rng = rng_factory()
+
+    jnp_fun = lambda lhs, rhs: np.linalg.lstsq(lhs, rhs)
+    onp_fun = lambda lhs, rhs: onp.linalg.lstsq(lhs, rhs)
+
+    a = rng(lhs_shape, dtype)
+    b = rng(rhs_shape, dtype)
+      
+    self.assertAllClose(jnp_fun(a, b), onp_fun(a, b), atol=1e-4, check_dtypes=True)
```

Comment:
NumPy always returns rank of `a` with `numpy.int32` type. Current JAX implementation returns `rank` with dtype `int32` or `int64` based on dtype of `a`. It makes sense not to check dtype for `rank`.

Reply:
Didn't know that, thanks!